File: read_xmltv.rb
require File.dirname(__FILE__) + '/lib/simple_pvr/pvr_initializer'
require File.dirname(__FILE__) + '/lib/simple_pvr/xmltv_reader'
require 'yaml'

if ARGV.length != 2
  puts "Requires two arguments: The XMLTV file name, and the channel mapping file name"
  exit 1
end

SimplePvr::PvrInitializer.setup

xmltv_file = File.new(ARGV[0])
mapping_to_channels = YAML.load_file(ARGV[1])

reader = SimplePvr::XmltvReader.new(SimplePvr::PvrInitializer.dao, mapping_to_channels)
reader.read(File.new(xmltv_file))

File: python/read_xmltv.py
import yaml
import codecs
import sys

from simple_pvr.pvr_initializer import pvr_initializer
from simple_pvr.xmltv_reader import XmltvReader

if len(sys.argv) != 3:
    print("Requires two arguments: The XMLTV file name, and the channel mapping file name")
    sys.exit(1)

pvr_initializer().setup

stream = file(sys.argv[2], 'r')
mapping_to_channels = yaml.load(stream)

reader = XmltvReader(mapping_to_channels)

xmltv_file = sys.argv[1]
reader.read(xmltv_file)